backend/cmm_helpers.mli

```diff
@@ -1226,3 +1226,149 @@ val set_field_unboxed :
 val dls_get : dbg:Debuginfo.t -> expression
 
 val poll : dbg:Debuginfo.t -> expression
+
+module Scalar_type : sig
+  (** A static_cast from a larger integral type to a smaller one logically truncates the
+      upper bits. Note that values are stored in registers sign- or zero- extended
+      according to their signdness, so the result may be sign-extended.
+
+      A static_cast from a smaller integral type to an equal or larger sized-integral type
+      sign- or zero-extends the input value according to the sign of the result.
+
+      A static_cast from an integral type to a float is pretty self-explanatory.
+
+      A static_cast from a float to an integral type always rounds toward zero. If the
+      resulting integral does not fit in the destination type, the result is unspecified
+      (although it's generally zero).
+
+      Casting floats to/from unsigned register-width integers is not implemented and will
+      raise in the compiler. *)
```

Comment:
This doc comment should be attached to that `static_cast` type declaration below, rather than the module itself, which would instead benefit from a high-level explanation of what is the purpose of this module (and its sub-modules).